crypto/fipsmodule/cipher/internal.h

```diff
@@ -73,12 +74,46 @@ extern "C" {
 // EVP_CIPH_MODE_MASK contains the bits of |flags| that represent the mode.
 #define EVP_CIPH_MODE_MASK 0x3f
 
+// Set of EVP_AEAD->evp_aead_id identifiers, zero is reserved as the "unknown"
+// value since it is the default for a structure.
+#define AEAD_UNKNOWN_ID 0
+#define AEAD_AES_128_CTR_HMAC_SHA256_ID 1
+#define AEAD_AES_256_CTR_HMAC_SHA256_ID 2
+#define AEAD_AES_128_GCM_SIV_ASM_ID 3
```

Comment:
I don't think you can have unique IDs for different implementations of the same algorithm. That is, for example  `AEAD_AES_128_GCM_SIV_ASM_ID` and `AEAD_AES_128_GCM_SIV_ID` i.e. the same AEAD encryption algorithm but with a different backend implementation; the former machine-optimised, the latter not.

Let's say you serialise an SSL object from a machine with machine-optimised code. One can imagine an SSL object transfer to either:
* a machine with a different CPU that the optimised code does not work on?
* a machine or libssl using a different version of libcrypto that doesn't even use machine-optimised code for some reason?

Wouldn't this break then?

Reply:
Discussed offline and will take the suggestion to not make these identifiers specific to the implementation.